gr-blocks/lib/nlog10_ff_impl.cc

```diff
@@ -51,11 +57,18 @@ namespace gr {
       const float *in = (const float *) input_items[0];
       float *out = (float *) output_items[0];
       int noi = noutput_items * d_vlen;
-      float n = d_n;
-      float k = d_k;
 
-      for (int i = 0; i < noi; i++)
-	out[i] = n * log10(std::max(in[i], (float) 1e-18)) + k;
+      for (int i = 0; i < noi; i++) {
+          out[i] = in[i] + d_float_min;
```

Comment:
That's not a solution! For example,`in[i] == -1.0f` is still negative after adding the minimum float.

No, we *have* to do `max(in,float_min)` in some way. I wish we had a VOLK kernel like `volk_32f_s32f_max_32f`, but we don't, do we, @n-west ? We can only compare two vectors of equal size and take the element-wise maximum.

Reply:
Now using `std::max()`.